sql/core/src/test/scala/org/apache/spark/sql/SetCommandSuite.scala

```diff
@@ -143,4 +144,15 @@ class SetCommandSuite extends QueryTest with SharedSparkSession with ResetSystem
       assert(!allValues.exists(v => v.contains(value1) || v.contains(value2)))
     }
   }
+
+  test("SPARK-42946: Set command could expose sensitive data through key") {
+    val key1 = "test.password"
+    val value1 = "test.value1"
+    withSQLConf(key1 -> value1) {
+      checkError(
+        intercept[ParseException](sql("SET ${test.password}")),
```

Comment:
It did not fail before this PR?

Reply:
Yes, as described in the PR desc. Well, if value1 here is in a invalid form, it may fail